### PR TITLE
Read value object from transform context as org.apache.avro.generic.GenericRecord only

### PR DESCRIPTION
**Pull request by aymkhalil**
Addressed feedback for the preview (#4):
* Remove the code that maps the object value from the transform context as GenericAvroRecord
* Avoid using optional class members on the FlattenStep
* Add unit tests for thrown exception

### CODE REVIEW

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java [hunk 1]
```diff
@@ -17,16 +17,15 @@
...
-import lombok.RequiredArgsConstructor;
+import lombok.Builder;
 import lombok.Value;
 import org.apache.avro.generic.GenericData;
 import org.apache.avro.generic.GenericRecord;
 import org.apache.pulsar.client.api.Schema;
 import org.apache.pulsar.common.schema.SchemaType;
 
-@RequiredArgsConstructor
+@Builder
```
Comment (cbornet) [inline pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java:28]: Very good idea ! We could do the same for all Steps

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java [hunk 2]
```diff
@@ -35,12 +34,12 @@ public class FlattenStep implements TransformStep {
 
   private static final String DEFAULT_DELIMITER = "_"; // '.' in not valid in AVRO field names
 
-  private final Optional<String> delimiter;
-  private final Optional<String> part;
+  @Builder.Default private String delimiter = DEFAULT_DELIMITER;
```
Comment (cbornet) [inline pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java:37]: can be final ?

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java [hunk 3]
```diff
@@ -35,12 +34,12 @@ public class FlattenStep implements TransformStep {
...
+  private String part;
```
Comment (cbornet) [inline pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java:38]: Can be final ?

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java [hunk 4]
```diff
@@ -191,7 +191,10 @@ public static CastStep newCastFunction(Map<String, Object> step) {
   }
 
   public static FlattenStep newFlattenFunction(Map<String, Object> step) {
-    return new FlattenStep(getStringConfig(step, "delimiter"), getStringConfig(step, "part"));
+    FlattenStep.FlattenStepBuilder builder = FlattenStep.builder();
+    getStringConfig(step, "part").map(part -> builder.part(part));
```
Comment (cbornet) [inline pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java:195]:
```suggestion
    getStringConfig(step, "part").ifPresent(builder::part);
```

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java [hunk 5]
```diff
@@ -191,7 +191,10 @@ public static CastStep newCastFunction(Map<String, Object> step) {
...
+    getStringConfig(step, "delimiter").map(delimiter -> builder.delimiter(delimiter));
```
Comment (cbornet) [inline pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java:196]:
```suggestion
    getStringConfig(step, "delimiter").ifPresent(builder::delimiter);
```

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java (hunk 3 above)
Comment (cbornet) [inline pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java:38]:
```suggestion
  private final String part;
```

**Comment by cbornet** [comment]
Did minor suggestions.
LGTM 👍

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java (hunk 3 above)
Comment (aymkhalil) [reply pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/FlattenStep.java:38]: it actually can.

**Pull request merged**